api/budget/api.go

```diff
@@ -173,6 +174,9 @@ func (c *client) doRequest(req interface{}, result interface{}) error {
 		}
 		resp, err = c.h.DoWithBody(req, bytes.NewReader(payload.Bytes()))
 		if err != nil {
+			if strings.HasSuffix(err.Error(), "Connection refused") {
+				return wireformat.NotAvailError{}
+			}
 			return errors.Annotate(err, "failed to execute request")
```

Comment:
This might be good enough to troubleshoot a connection failure with. It would certainly differentiate it from a 503.